### PR TITLE
Load regions from OCI gem

### PR DESCRIPTION
**Pull request by agrare**
The original list of regions wasn't sorted so this diff looks worse than it is, here is the diff from the sorted list:
```
+sa-santiago-1:
+  :name: sa-santiago-1
+  :hostname: https://sa-santiago-1.oraclecloud.com
 sa-saopaulo-1:
   :name: sa-saopaulo-1
   :hostname: https://sa-saopaulo-1.oraclecloud.com
+sa-vinhedo-1:
+  :name: sa-vinhedo-1
+  :hostname: https://sa-vinhedo-1.oraclecloud.com
+uk-cardiff-1:
+  :name: uk-cardiff-1
+  :hostname: https://uk-cardiff-1.oraclecloud.com
```

Depends on:
- [x] https://github.com/ManageIQ/manageiq/pull/21578

### CODE REVIEW

File: lib/tasks_private/oracle_cloud.rake [hunk 1]
```diff
@@ -0,0 +1,20 @@
+namespace :oracle_cloud do
+  namespace :regions do
+    desc "Update list of regions"
+    task :update => :environment do
+      require "oci"
+
+      regions = OCI::Regions::REGION_ENUM.map do |name|
```
Comment (Fryguy) [inline lib/tasks_private/oracle_cloud.rake:7]: Is this hardcoded into the gem?  If so, we should not generate the yaml at all and instead override the base method to just to this directly

**Comment by Fryguy** [comment]
Out of curiosity, where are these hostnames even used?  The URLs don't seem to be valid, and are built different from, say, get_service_endpoint:

```ruby
[2] pry(main)> OCI::Regions.get_service_endpoint("ap-chiyoda-1", :IdentityClient)
=> "https://identity.ap-chiyoda-1.oraclecloud8.com"
```

**Comment by agrare** [comment]
> Out of curiosity, where are these hostnames even used?

Most of the providers add one in, not sure where they are used or even if they are.

> The URLs don't seem to be valid, and are built different from, say, get_service_endpoint:

Yeah OCI only provides an interface to get the endpoint for each service, not just the region.  I removed the hostname for now we can follow-up with other providers and see if they're used at all by any

**Comment by Fryguy** [comment]
This looks good to me.  I think as a follow up, what we might want to do is change the core method `from_file` to `from_source`.  Then, this method wouldn't have to override `regions`, but would instead override `from_source` with 

```ruby
      def self.from_source
        require "oci"
        OCI::Regions::REGION_ENUM
          .map      { |name| {:name => name} }
          .index_by { |r| r[:name] }
      end
```

**Review by Fryguy — APPROVED** [review]

**Pull request merged**